app/component/departure/DepartureListContainer.js

```diff
@@ -0,0 +1,193 @@
+import Departure from './Departure';
+import filter from 'lodash/filter';
+import get from 'lodash/get';
+import moment from 'moment';
+import Link from 'react-router/lib/Link';
+import cx from 'classnames';
+import connectToStores from 'fluxible-addons-react/connectToStores';
+
+const mergeDepartures = departures =>
+  Array.prototype.concat.apply([], departures).sort((a, b) => a.stoptime - b.stoptime);
+
+const asDepartures = stoptimes =>
+  stoptimes.map(pattern =>
+    pattern.stoptimes.map(stoptime => {
+      const isArrival = stoptime.pickupType === 'NONE';
+      const canceled = stoptime.realtimeState === 'CANCELED' ||
+        window.mock && stoptime.realtimeDeparture % 40 === 0;
+      const arrivalTime = stoptime.serviceDay +
+        (canceled
+          ? stoptime.realtimeArrival
+          : stoptime.scheduledArrival);
+      const departureTime = stoptime.serviceDay +
+        (canceled
+          ? stoptime.realtimeDeparture
+          : stoptime.scheduledDeparture);
+      const stoptimeTime = isArrival ? arrivalTime : departureTime;
+
+      return {
+        canceled,
+        isArrival,
+        stoptime: stoptimeTime,
+        stop: stoptime.stop,
+        realtime: stoptime.realtime,
+        pattern: pattern.pattern,
+        trip: { gtfsId: stoptime.trip.gtfsId },
+      };
+    })
+  );
+
+class DepartureListContainer extends Component {
+  static propTypes = {
+    rowClasses: PropTypes.string.isRequired,
+    stoptimes: PropTypes.array.isRequired,
+    currentTime: PropTypes.object.isRequired,
+    limit: PropTypes.number,
+    infiniteScroll: PropTypes.bool,
+    showStops: PropTypes.bool,
+    routeLinks: PropTypes.bool,
+    className: PropTypes.string,
+  };
+
+  onScroll = () => {
+    if (this.props.infiniteScroll && typeof window !== 'undefined' && window !== null) {
+      return this.scrollHandler;
+    }
+    return null;
+  };
+
+  getDepartures(rowClasses) {
+    const departureObjs = [];
+    const currentTime = this.props.currentTime.unix();
+    let currentDate = this.props.currentTime.clone()
+      .startOf('day')
+      .unix();
+    let tomorrow = this.props.currentTime.clone()
+      .add(1, 'day')
+      .startOf('day')
+      .unix();
+
+    const departures = mergeDepartures(asDepartures(this.props.stoptimes))
+      .filter(departure => currentTime < departure.stoptime).slice(0, this.props.limit);
+    for (let departure of departures) {
+      if (departure.stoptime >= tomorrow) {
+        departureObjs.push(
+          <div
+            key={moment(departure.stoptime * 1000).format('DDMMYYYY')}
+            className="date-row border-bottom"
+          >
+            {moment(departure.stoptime * 1000).format('dddd D.M.YYYY')}
+          </div>);
+
+        currentDate = tomorrow;
+        tomorrow = moment.unix(currentDate)
+          .add(1, 'day')
+          .startOf('day')
+          .unix();
+      }
+
+      const id = `${departure.pattern.code} : ${departure.stoptime}`;
```

Comment:
Are the space around the colon added in purpose? Coffee version didn't have them.